Firebase.agent.lib.nut

```diff
@@ -641,24 +655,57 @@ class Firebase {
                 }
                 if (200 <= res.statuscode && res.statuscode < 300) {
                     onSuccess(data);
+                    _tooManyReqTimer = false;
+                } else if (res.statuscode == 28 || res.statuscode == 429) {
+                    local now = time();
+                    // Too many requests, set _tooManyReqTimer to prevent more requests to FB
+                    if (_tooManyReqTimer == false) {
+                        // This is the first 429 we have seen set a default timeout
+                        _tooManyReqTimer = now + DEFAULT_BACK_OFF_TIMEOUT_SEC;
+                    } else if (_tooManyReqTimer <= now) {
+                        // Firebase is still overwhelmed after first timeout expired, 
+                        // Let's block requests for longer to let FB recover
+                        _tooManyReqCounter += 1
+                        _tooManyReqTimer = now + (DEFAULT_BACK_OFF_TIMEOUT_SEC * _tooManyReqCounter);
```

Comment:
`_tooManyReqTimer = now + (DEFAULT_BACK_OFF_TIMEOUT_SEC * _tooManyReqCounter++);` and the prev line can be removed